saleae/saleae.py

```diff
@@ -37,15 +37,44 @@
 PY3K = sys.version_info[0] == 3
 
 @enum.unique
-class Trigger(enum.IntEnum):
-	'''Trigger types to start sampling.'''
+class TriggerType(enum.IntEnum):
+	'''trigger types to start sampling.'''
 	# Python convention is to start enums at 1 for truth checks, but it
 	# seems reasonable that no trigger should compare as false
 	NoTrigger = 0
 	High = 1
 	Low = 2
 	Posedge = 3
 	Negedge = 4
+	Pospulse = 5
+	Negpulse = 6
+
+class Trigger():
+	'''
+	:param trigger_type: See TriggerType class
+	
+	:Keyword Arguments:
+		minimum_pulse_width (String): valid only for Pospulse/Negpulse trigger types (refer to Saleae for allowed ranges; default = '0.1')
+		maximum_pulse_width (String): valid only for Pospulse/Negpulse trigger types (refer to Saleae for allowed ranges; default = '0.1')
+
+	>>> Trigger(TriggerType.Pospulse, '0.0001', '0.0002')
+	>>> Trigger(5, '0.0001', '0.0002')
+	'''
+	def __init__(self, trigger_type, **kwargs):
+		self.trigger_type = TriggerType(trigger_type)
+
+		if self.trigger_type == TriggerType.Pospulse or self.trigger_type == TriggerType.Negpulse:
+			self.minimum_pulse_width = kwargs.get('minimum_pulse_width', '0.1')
+			self.maximum_pulse_width = kwargs.get('maximum_pulse_width', '0.1')
```

Comment:
Well, in case the **kwargs are left empty while the user defines a Pos/Neg-pulse, the min/max widths will default to '0.1'. I've checked it with logic and the values are set to 0 and 1, so I can change the defaults, which would be more reasonable in my opinion.